src/herder/TransactionQueue.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "crypto/SecretKey.h"
+#include "herder/TxSetFrame.h"
+#include "transactions/TransactionFrame.h"
+#include "util/HashOfHash.h"
+#include "util/XDROperators.h"
+#include "xdr/Stellar-transaction.h"
+
+#include <deque>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+namespace medida
+{
+class Counter;
+}
+
+namespace stellar
+{
+
+class Application;
+
+struct AccountTransactionsQueueState
```

Comment:
Forgot to address those couple points:

this type probably belongs under the TransactionQueue namespace as it doesn't make sense in the stellar namespace

Also, AccountTransactionsState doesn't really mean anything: it's not a state, I think it's more like AccountQueueInformation or something like that